src/box/txn_limbo.c

```diff
@@ -490,6 +490,87 @@ txn_limbo_submit(struct txn_limbo *limbo, uint32_t id, struct txn *txn,
 	return 0;
 }
 
+int
+txn_limbo_flush(struct txn_limbo *limbo)
+{
+	/* Fast path. */
+	if (txn_limbo_is_empty(limbo))
+		return 0;
+	if (txn_limbo_last_entry(limbo)->state != TXN_LIMBO_ENTRY_VOLATILE)
+		return 0;
+	/*
+	 * Slow path.
+	 * The limbo queue guarantees that if a txn is trying to be submitted
+	 * into it, then the submission would return right after all the
+	 * previous txns are sent to the journal and before any newer txns do
+	 * the same.
```

Comment:
This won't be true, if it's not guaranteed, that transactions are woken up in the same order, they entered the wating. I'm afraid of the following situation:

1. Limbo is full and non-empty, 2 async transactions are waiting for space in limbo
2. The confim is written, we're having 2 async tx in limbo in `VOLATILE` and then the second one is woken up before the first one
3. If the second transaction is the one, which is used here for flush, then you won't have all txns in journal queue

Why the second point is not possible and what am I missing